src/test/java/org/commcare/formplayer/tests/SnapshotTests.java

```diff
@@ -0,0 +1,110 @@
+
+import org.apache.commons.io.FileUtils;
+import org.commcare.formplayer.sandbox.UserSqlSandbox;
+import org.commcare.formplayer.sqlitedb.UserDB;
+import org.commcare.formplayer.utils.GenerateSnapshotDatabases;
+import org.commcare.formplayer.utils.TestContext;
+import org.javarosa.core.services.storage.IMetaData;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.util.Iterator;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Tests the compatibility of the current code runtime against the currently captured database
+ * snapshots.
+ *
+ * If these tests fail, the snapshots need to be rebuild and databases will need to be cold started
+ * in any existing environments after updating.
+ *
+ * @author ctsims
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestContext.class)
+public class SnapshotTests extends BaseTestClass {
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        copySnapshotResources();
+    }
+
+    private void copySnapshotResources() {
+        try {
+            File destDirectory = new File(getDatabaseFolderRoot());
+
+            FileUtils.copyDirectory(new File(GenerateSnapshotDatabases.snapshotDbDirectory)
+                    , destDirectory);
+        } catch (IOException exception) {
+            exception.printStackTrace();
+            Assert.fail("Error moving snapshot data to test directory");
+        }
+    }
+
+    protected boolean remoteDatabaseFoldersAfterTests() {
+        return true;
+    }
+
+
+    @Test
+    public void testUserSandbox() throws Exception {
+        UserDB database = getUserDbConnector("snapshot", "snapshot_test", null);
+        if (!database.databaseFileExists()) {
+            Assert.fail("Snapshot UserDB Missing for tests you may need to rebuild the snapshot " +
+                    "with the CreateSnapshotDbs Gradle Task ");
+        }
+        //Try to enumerate records of each type
+        UserSqlSandbox sandbox = new UserSqlSandbox(database);
+        enumerate("Case", sandbox.getCaseStorage());
+        enumerate("Ledger", sandbox.getLedgerStorage());
+        enumerate("User",sandbox.getUserStorage());
+        enumerate("App Fixtures",sandbox.getAppFixtureStorage());
+        enumerate("User Fixtures",sandbox.getUserFixtureStorage());
+    }
+
+    @Test
+    public void testAppSandbox() throws Exception {
+        try {
+            //Note - Currently this is actually expected to blow away the DB a surprisingly large
+            //percentage of the time, but should be expanded in the future to be more robust
+            //when app installs are less fragile
+            doInstall("sandbox_reference/snapshot.json");
```

Comment:
Since this is not relevant today as we wipe out the App DB on any errors, checking if this test doesn't serve a purpose today and is only here to better support any changes in future with how we handle App DB.

Reply:
Hey @shubham1g5 ,

Confirming that right now that this wouldn't really catch "Breaking changes" in the appdb schema, the big thing it would catch today is a change to the App DB wiping process you are describing that results in a broken state, which I think is a realistic regression. The more I dig into the app db install process the more fragile it looks, so I wanted to make sure that starting with an old DB in place works with each release.